src/controllers/confirm.company.controller.ts

```diff
@@ -11,21 +11,50 @@ import {
   EligibilityStatusCode
 } from "private-api-sdk-node/dist/services/confirmation-statement";
 import { TRADING_STATUS_PATH, urlParams } from "../types/page.urls";
+import { isInFuture, toReadableFormat } from "../utils/date";
+import { DateTime } from "luxon";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
   try {
-    const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
-    return res.render(Templates.CONFIRM_COMPANY, { company, templateName: Templates.CONFIRM_COMPANY });
+    const companyProfile: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
+    return res.render(Templates.CONFIRM_COMPANY, buildPageOptions(companyProfile));
   } catch (e) {
-    next(e);
+    return next(e);
+  }
+};
+
+const buildPageOptions = (companyProfile: CompanyProfile): Object => {
+  // need to extract the nextMadeUpTo before company profile is formatted for display as it will modify the date format
+  const nextMadeUpTo = companyProfile?.confirmationStatement?.nextMadeUpTo;
+
+  const pageOptions = {
+    company: formatForDisplay(companyProfile),
+    templateName: Templates.CONFIRM_COMPANY
+  };
+
+  if (!isFilingDue(nextMadeUpTo)) {
+    pageOptions["notDueWarning"] = {
+      todaysDate: toReadableFormat(DateTime.now().toISODate())
+    };
   }
+  return pageOptions;
+};
+
+const isFilingDue = (nextMadeUpTo: string | undefined): boolean => {
+  let due = true;
+  if (nextMadeUpTo) {
+    if (isInFuture(nextMadeUpTo)) {
+      due = false;
+    }
+  }
+  return due;
 };
 
 export const post = async (req: Request, res: Response, next: NextFunction) => {
-  const session: Session = req.session as Session;
-  const companyNumber = req.query["companyNumber"] as string;
-  const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
   try {
+    const session: Session = req.session as Session;
```

Comment:
is this session variable actually used?

Reply:
looks like it is passed into the eligibility service and is used to create the OAuth API Client in there to make the api call.
It was already there but I noticed those 3 lines were outside the try {} so if they threw an error then page would hang. So moved them inside the try{}.